File: lectures/01/ArrayExample.java
// Example of run-time bounds checking for array access
// (Compare with ArrayExample.c)

class ArrayExample {
  public static void main(String[] args) {

    int[] x = new int[10];

    for (int i = 0; i < 10000; ++i) {
      System.out.printf("%5d: %d%n", i, x[i]);
    }
  }
}

File: lectures/01/ArrayExample.c
/* Example of lack of run-time bounds checking for array access */

#include <stdio.h>

int main()
{
  int x[10], i;

  for (i = 0; i < 10000; ++i) {
    printf("%5d: %d\n", i, x[i]);
  }

  return 0;
}
